frontend/src/routes/wallet/keypair/taint.tsx

```diff
@@ -0,0 +1,135 @@
+import { useState, useCallback } from 'react'
+import { Intent } from '../../../config/intent'
+import { Header } from '../../../components/header'
+import { Button } from '../../../components/button'
+import { AppToaster } from '../../../components/toaster'
+import { BreakText } from '../../../components/break-text'
+import { KeyValueTable } from '../../../components/key-value-table'
+import { useCurrentKeypair } from '../../../hooks/use-current-keypair'
+import { requestPassphrase } from '../../../components/passphrase-modal'
+import { updateKeyPairAction } from '../../../contexts/global/global-actions'
+import { useGlobal, GlobalDispatch } from '../../../contexts/global/global-context'
+import { createLogger } from '../../../lib/logging'
+import { Service } from '../../../service'
+
+const logger = createLogger('Taint')
+
+const useTaint = (dispatch: GlobalDispatch, pubKey?: string, wallet?: string) => {
+  const [loading, setLoading] = useState(false);
+
+  const taint = useCallback(
+    async () => {
+      setLoading(true);
+      try {
+        if (!pubKey || !wallet) {
+          return
+        }
+
+        const passphrase = await requestPassphrase()
```

Comment:
suggestion (non blocking): I've been trying to keep all the biz logic in action creators where possible. Could we create an async `taintKeypairAction`?

I realise that there are currently some places where biz logic is in hooks and/or components but I'd like to refactor them when possible.

Reply:
i have seen this pattern in some of the cases, and i find it extremely confusing.

my gut feeling is that it's just way too much UI logic to trigger react components directly jumping in and out from a global action, also it reduces how reusable these actions can be. it would totally be a lot more transparent if this would happen by a state update, but we're calling requestPassphrase right from an action feels just super weird. 

right now this would be possible, because this is the only component which allows updating the taint state. which raises the question how global this update flow really is. not sure, probably a million different ways to do it, i can also just add it in there tbh :D

Reply:
Okay lets chat about this! Happy for this as is for now